### PR TITLE
Force all our dialogs into the application theme, forcibly

### LINKED ISSUE #3654: About modal's context menu is not legible when using dark theme
# Environment

```none
Windows build number: 10.0.18363.476
Windows Terminal version (if applicable): 0.6.2951.0
```

# Steps to reproduce

1. Make sure Windows Terminal is in dark theme and Windows 10 default app mode is light
2. Open `about` modal
3. Highlight some text and right click

# Expected behavior
Context menu should be legible with white text on dark background.

# Actual behavior

Context menu text is not legible.

Screenshot (35)

### LINKED ISSUE #5195: About and Close all dialogs do not correctly show all UI elements
# Environment

```none
Windows build number: [10.0.18363.720]
Windows Terminal version: latest unreleased build from master

Any other software? No
```

# Steps to reproduce

1. Windows default app mode: light
2. Windows terminal requestedTheme: dark
3. Open **About** or **Close all** dialog
4. Observe not all graphical elements are visible
  a. On the **About** dialog, hover over a hyperlink and observe it disappear
  b. On the **Close all** dialog, observe that the **Cancel** button is not visible (but is active because it Cancels when clicking the area where the button is located.

# Expected behavior

1. **About** dialog: Can see the hyperlink when hovering over it with mouse cursor
2. **Close all** dialog: Can see the **Cancel** button.

# Actual behavior

1. **About** dialog: Cannot see the hyperlink when hovering over it with mouse cursor

help

2. **Close all** dialog: Cannot see the **Cancel** button.

close-all

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
Because we cannot set RequestedTheme at the application level, we
occasionally run into issues where parts of our UI end up themed
incorrectly.  Dialogs, for example, live under a different Xaml root
element than the rest of our application. This makes our popup menus and
buttons "disappear" when the user wants Terminal to be in a different
theme than the rest of the system.  This hack---and it _is_ a
hack--walks up a dialog's ancestry and forces the theme on each element
up to the root. We're relying a bit on Xaml's implementation details
here, but it does have the desired effect.

It's not enough to set the theme on the dialog alone.

Fixes #3654.
Fixes #5195.

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
This introduces an issue where the first time a dialog is displayed its theme is locked, and if you change the theme at runtime and reopen the dialog it doesn't change.

I honestly do not care, because that is more consistent than what we had before. :P

**Comment by DHowett-MSFT** [comment]
The search box lives inside the TermControl, not in an alternate xaml root object

File: src/cascadia/TerminalApp/AppLogic.cpp [hunk 1]
```diff
@@ -260,7 +260,27 @@ namespace winrt::TerminalApp::implementation
...
+        // GH#5195, GH#3654 Because we cannot set RequestedTheme at the application level,
+        // we occasionally run into issues where parts of our UI end up themed incorrectly.
+        // Dialogs, for example, live under a different Xaml root element than the rest of
+        // our application. This makes our popup menus and buttons "disappear" when the
+        // user wants Terminal to be in a different theme than the rest of the system.
+        // This hack---and it _is_ a hack--walks up a dialog's ancestry and forces the
+        // theme on each element up to the root. We're relying a bit on Xaml's implementation
+        // details here, but it does have the desired effect.
+        // It's not enough to set the theme on the dialog alone.
+        auto themingLambda{ [this](const Windows::Foundation::IInspectable& sender, const RoutedEventArgs&) {
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/AppLogic.cpp:272]: Do we need to do weak_ref magic on `this` here?

**Comment by AnuthaDev** [comment]
@DHowett-MSFT I think this PR introduced a new problem that the about dialog no longer opens and closes smoothly with an animation, it appears and disappears abruptly, also the dialog border color is greyish in light theme instead of the accent color (maybe its intended).

**Comment by DHowett-MSFT** [comment]
Good catch. The dialog appearing abruptly is better than not having visible menus and buttons, though, so I'm going to file a followup issue and let it ride until post-1.0

**Pull request merged**